cloudinit/distros/debian.py

```diff
@@ -66,10 +66,21 @@ def get_locale(self):
         if not self.system_locale:
             self.system_locale = read_system_locale()
 
-        # Return system_locale setting if valid, else use default locale
-        return (
-            self.system_locale if self.system_locale else self.default_locale
-        )
+        if self.system_locale:
+            LOG.debug(
+                "System locale set to %s via %s",
+                self.system_locale,
+                LOCALE_CONF_FN,
+            )
+            return self.system_locale
+        else:
+            LOG.debug(
```

Comment:
Changeset looks good, but we have no unittest coverage for this path of the conditional

Reply:
tested on images w/out /etc/default/locale. Success regenerating locales in absence of system defaults

Reply:
Looking at this more, I may move the logging around. In the case where no locale has been specified, `get_locale` is going to get called twice so we'll see duplicate logs